Humble diagnostic_common_diagnostics build error

For some reason `package.xml` of this package has `catkin` `buildtool` tag that seems off for a ROS 2 package. It produces error such as 

```
The manifest of package "diagnostic_common_diagnostics" (with format version 1) must not contain the following tags: exec_depend.
Either update to a newer format or replace <exec_depend> tags with <run_depend> tags.
```

I think the correct way is just like `foxy` and `galactic` branch meaning the `buildtool` tag must be `ament_cmake` and `ament_cmake_python`. A pull request #261 have been opened by another user as well that fixes it.

@ct2034 Even I am facing the same issue

```
--- stderr: diagnostic_common_diagnostics                                                                          
Error parsing '/home/sumedh/mav_platform_r2/src/diagnostics/diagnostic_common_diagnostics/package.xml':
Traceback (most recent call last):
  File "/opt/ros/humble/share/ament_cmake_core/cmake/core/package_xml_2_cmake.py", line 150, in <module>
    main()
  File "/opt/ros/humble/share/ament_cmake_core/cmake/core/package_xml_2_cmake.py", line 53, in main
    raise e
  File "/opt/ros/humble/share/ament_cmake_core/cmake/core/package_xml_2_cmake.py", line 49, in main
    package = parse_package_string(
  File "/usr/lib/python3/dist-packages/catkin_pkg/package.py", line 786, in parse_package_string
    raise InvalidPackage('Error(s):%s' % (''.join(['\n- %s' % e for e in errors])), filename)
catkin_pkg.package.InvalidPackage: Error(s) in package '/home/sumedh/mav_platform_r2/src/diagnostics/diagnostic_common_diagnostics/package.xml':
Error(s):
- The manifest of package "diagnostic_common_diagnostics" (with format version 1) must not contain the following tags: exec_depend
- Either update to a newer format or replace <exec_depend> tags with <run_depend> tags.
CMake Error at /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_package_xml.cmake:95 (message):
  execute_process(/usr/bin/python3.8
  /opt/ros/humble/share/ament_cmake_core/cmake/core/package_xml_2_cmake.py
  /home/sumedh/mav_platform_r2/src/diagnostics/diagnostic_common_diagnostics/package.xml
  /home/sumedh/mav_platform_r2/build/diagnostic_common_diagnostics/ament_cmake_core/package.cmake)
  returned error code 1
Call Stack (most recent call first):
  /opt/ros/humble/share/ament_cmake_core/cmake/core/ament_package_xml.cmake:49 (_ament_package_xml)
  /opt/ros/humble/share/ament_cmake_python/cmake/ament_python_install_package.cmake:60 (ament_package_xml)
  /opt/ros/humble/share/ament_cmake_python/cmake/ament_python_install_package.cmake:39 (_ament_cmake_python_install_package)
  CMakeLists.txt:10 (ament_python_install_package)

```

--- a/diagnostic_common_diagnostics/diagnostic_common_diagnostics/ntp_monitor.py
+++ b/diagnostic_common_diagnostics/diagnostic_common_diagnostics/ntp_monitor.py
@@ -32,22 +32,21 @@
 # ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
 
+import re
+import socket
+import sys
+import threading
+from subprocess import PIPE, Popen, TimeoutExpired
+
+import diagnostic_updater as DIAG
 import rclpy
 from rclpy.node import Node
 
-import diagnostic_updater as DIAG
-
-import sys
-import threading
-import socket
-from subprocess import Popen, PIPE, TimeoutExpired
-import re
-
 
 class NTPMonitor(Node):
-    
+
     def __init__(self, ntp_hostname, offset=500, self_offset=500,
-                 diag_hostname = None, error_offset = 5000000,
+                 diag_hostname=None, error_offset=5000000,
                  do_self_test=True):
         super().__init__(__class__.__name__)
 
@@ -57,27 +56,29 @@ class NTPMonitor(Node):
         self.diag_hostname = diag_hostname
         self.error_offset = error_offset
         self.do_self_test = do_self_test
-        
+
         self.hostname = socket.gethostname()
         if self.diag_hostname is None:
             self.diag_hostname = self.hostname
 
         self.stat = DIAG.DiagnosticStatus()
         self.stat.level = DIAG.DiagnosticStatus.OK
-        self.stat.name = "NTP offset from "+ self.diag_hostname + " to " + self.ntp_hostname
+        self.stat.name = "NTP offset from " + \
+            self.diag_hostname + " to " + self.ntp_hostname
         self.stat.message = "OK"
         self.stat.hardware_id = self.hostname
         self.stat.values = []
 
         self.self_stat = DIAG.DiagnosticStatus()
         self.self_stat.level = DIAG.DiagnosticStatus.OK
-        self.self_stat.name = "NTP self-offset for "+ self.diag_hostname
+        self.self_stat.name = "NTP self-offset for " + self.diag_hostname
         self.self_stat.message = "OK"
         self.self_stat.hardware_id = self.hostname
         self.self_stat.values = []
 
         self.mutex = threading.Lock()
-        self.pub = self.create_publisher(DIAG.DiagnosticArray, "/diagnostics", 10)
+        self.pub = self.create_publisher(
+            DIAG.DiagnosticArray, "/diagnostics", 10)
 
         # we need to periodically republish this
         self.current_msg = None
@@ -91,7 +92,6 @@ class NTPMonitor(Node):
 
     def checkCB(self):
         new_msg = DIAG.DiagnosticArray()
-        # new_msg.header.stamp = rospy.get_rostime()
 
         st = self.ntp_diag(self.stat)
         if st is not None:
@@ -105,23 +105,22 @@ class NTPMonitor(Node):
         with self.mutex:
             self.current_msg = new_msg
 
-
-
     def ntp_diag(self, st):
         """Adds ntp diagnostics to the given status message and returns the new status
 
         Args:
             st (DIAG.DiagnosticStatus()): The diagnostic status object to populate
         """
-        
+
         def add_kv(stat_values, key, value):
             kv = DIAG.KeyValue()
             kv.key = key
             kv.value = value
             stat_values.append(kv)
-            
+
         PROCESS_NAME = "ntpdate"
-        p = Popen([PROCESS_NAME, "-q", self.ntp_hostname], stdout=PIPE, stdin=PIPE, stderr=PIPE)
+        p = Popen([PROCESS_NAME, "-q", self.ntp_hostname],
+                  stdout=PIPE, stdin=PIPE, stderr=PIPE)
         try:
             stdout, stderr = p.communicate(timeout=5)
         except TimeoutExpired:
@@ -129,13 +128,14 @@ class NTPMonitor(Node):
             stdout, stderr = p.communicate()
             self.get_logger().error('Timeout running %s: "%s"' % (PROCESS_NAME, stderr))
         rc = p.returncode
-        
+
         st.values = []
         add_kv(st.values, "Offset tolerance (us)", str(self.offset))
-        add_kv(st.values, "Offset tolerance (us) for Error", str(self.error_offset))
+        add_kv(st.values, "Offset tolerance (us) for Error",
+               str(self.error_offset))
         output = stdout.decode("UTF-8")
         errors = stderr.decode("UTF-8")
-        
+
         if rc == 0:
             re_match = re.search("offset (.*),", output)
             measured_offset = float(re_match[1])*1000000
@@ -161,33 +161,32 @@ class NTPMonitor(Node):
         return st
 
 
-
 def ntp_monitor_main(argv=sys.argv[1:]):
     import argparse
     parser = argparse.ArgumentParser()
     parser.add_argument("--ntp_hostname",
-                      action="store", default="pool.ntp.org",
-                      type=str)
+                        action="store", default="pool.ntp.org",
+                        type=str)
     parser.add_argument("--offset-tolerance", dest="offset_tol",
-                      action="store", default=500,
-                      help="Offset from NTP host", metavar="OFFSET-TOL",
-                      type=int)
+                        action="store", default=500,
+                        help="Offset from NTP host", metavar="OFFSET-TOL",
+                        type=int)
     parser.add_argument("--error-offset-tolerance", dest="error_offset_tol",
-                      action="store", default=5000000,
-                      help="Offset from NTP host. Above this is error", metavar="OFFSET-TOL",
-                      type=int)
+                        action="store", default=5000000,
+                        help="Offset from NTP host. Above this is error", metavar="OFFSET-TOL",
+                        type=int)
     parser.add_argument("--self_offset-tolerance", dest="self_offset_tol",
-                      action="store", default=500,
-                      help="Offset from self", metavar="SELF_OFFSET-TOL",
-                      type=int)
+                        action="store", default=500,
+                        help="Offset from self", metavar="SELF_OFFSET-TOL",
+                        type=int)
     parser.add_argument("--diag-hostname", dest="diag_hostname",
-                      help="Computer name in diagnostics output (ex: 'c1')",
-                      metavar="DIAG_HOSTNAME",
-                      action="store", default=None,
-                      type=str)
+                        help="Computer name in diagnostics output (ex: 'c1')",
+                        metavar="DIAG_HOSTNAME",
+                        action="store", default=None,
+                        type=str)
     parser.add_argument("--no-self-test", dest="do_self_test",
-                      help="Disable self test",
-                      action="store_false", default=True)
+                        help="Disable self test",
+                        action="store_false", default=True)
     args = parser.parse_args(args=argv)
 
     offset = args.offset_tol
@@ -200,17 +199,19 @@ def ntp_monitor_main(argv=sys.argv[1:]):
 
     rclpy.spin(ntp_monitor)
 
+
 def main(args=None):
     rclpy.init(args=args)
     try:
         ntp_monitor_main()
-    except KeyboardInterrupt: pass
-    except SystemExit: pass
+    except KeyboardInterrupt:
+        pass
+    except SystemExit:
+        pass
     except:
         import traceback
         traceback.print_exc()
 
+
 if __name__ == "__main__":
     main()
-    
-

--- a/diagnostic_common_diagnostics/package.xml
+++ b/diagnostic_common_diagnostics/package.xml
@@ -1,18 +1,22 @@
-<package>
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
   <name>diagnostic_common_diagnostics</name>
   <version>1.9.3</version>
   <description>diagnostic_common_diagnostics</description>
-  <author email="brice.rebsamen@gmail.com">Brice Rebsamen</author>
+  <!-- author -->
+  <maintainer email="brice.rebsamen@gmail.com">Brice Rebsamen</maintainer>
+  <!-- maintainers -->
   <maintainer email="namniart@gmail.com">Austin Hendrix</maintainer>
   <maintainer email="brice.rebsamen@gmail.com">Brice Rebsamen</maintainer>
   <maintainer email="ralph.lange@de.bosch.com">Ralph Lange</maintainer>
 
-  <license>BSD</license>
+  <license>BSD-3-Clause</license>
 
   <url type="website">http://ros.org/wiki/diagnostic_common_diagnostics</url>
-<!-- <url type="bugtracker"></url> -->
 
-  <buildtool_depend>catkin</buildtool_depend>
+  <buildtool_depend>ament_cmake</buildtool_depend>
+  <buildtool_depend>ament_cmake_python</buildtool_depend>
 
   <exec_depend>rclpy</exec_depend>
   <exec_depend>diagnostic_updater</exec_depend>
@@ -29,6 +33,6 @@
   <test_depend>ament_cmake_lint_cmake</test_depend>
 
   <export>
-    <architecture_independent/>
+    <build_type>ament_cmake</build_type>
   </export>
 </package>

